File: app/src/main/java/io/github/fanky10/android/mvpunittest/mvp/LoginContract.java
package io.github.fanky10.android.mvpunittest.mvp;

public interface LoginContract {
    interface Model {
        boolean authenticateUser(String username, String password);
    }

    interface View {
        void init();
        String getUsername();
        String getPassword();
        void showOnAuthenticationErrorMessage();
        void showWelcomeScreen();
        void showForgotPasswordScreen();
    }

    interface Presenter {
        void init();
        void onPause();
        void onResume();
        void onLoginPressed();
        void onForgotPasswordPressed();
    }
}

File: app/src/main/java/io/github/fanky10/android/mvpunittest/mvp/LoginContract.kt
package io.github.fanky10.android.mvpunittest.mvp

interface LoginContract {
    interface Model {
        fun authenticateUser(username: String, password: String): Boolean
    }

    interface View {
        val username: String
        val password: String
        fun init()
        fun showOnAuthenticationErrorMessage()
        fun showWelcomeScreen()
        fun showForgotPasswordScreen()
    }

    interface Presenter {
        fun init()
        fun onPause()
        fun onResume()
        fun onLoginPressed()
        fun onForgotPasswordPressed()
    }
}
